ledger/ledger-on-sql/src/main/scala/com/daml/ledger/on/sql/queries/PostgresqlQueries.scala

```diff
@@ -30,7 +30,7 @@ final class PostgresqlQueries(override protected implicit val connection: Connec
   }
 
   override protected val updateStateQuery: String =
-    s"INSERT INTO $StateTable VALUES ({key}, {value}) ON CONFLICT(key) DO UPDATE SET value = {value}"
+    s"INSERT INTO $StateTable (key, key_hash, value) VALUES ({key}, {key_hash}, {value}) ON CONFLICT(key_hash) DO UPDATE SET value = {value}"
```

Comment:
Technically this would corrupt the database on a hash collision (where the `key_hash` value conflicts with an existing row, but the `key` values are different).

Are we worried about that? If yes, you could do something like `ON CONFLICT(key_hash) DO UPDATE SET value = {value} WHERE EXCLUDED.key = {key}` and then count the number of affected rows (should be exactly 1).

Reply:
The `key_hash` is generated with the SHA256 algorithm, so I would assume that we're safe here but a green light from you @rautenrieth-da or @mziolekda would be great :)